ocp_utilities/infra.py

```diff
@@ -374,3 +373,28 @@ def create_update_secret(
     secret.data_dict = {secret_key: dict_base64_encode(_dict=secret_data_dict)}
 
     return secret.deploy()
+
+
+def get_pod_by_name_prefix(
+    dyn_client: DynamicClient, pod_prefix: str, namespace: str, get_all: bool = False
+) -> Pod | List[Pod]:
```

Comment:
I don't like it when we return two types, we need to return only one type.
I think it's better to always return list.

also I don't see any reason to return only one pod (the first one), we should return all pods and the user can take the first or use them all

Reply:
done